tools/run-integration-tests.py

```diff
@@ -0,0 +1,183 @@
+                env = os.environ.copy()
+
+                # Add check_version variable
+                if check_version:
+                    env["UACLIENT_BEHAVE_CHECK_VERSION"] = check_version
+
+                # If not told to run from proposed, run from local changes
+                if proposed:
+                    env["UACLIENT_BEHAVE_ENABLE_PROPOSED"] = "1"
+                else:
+                    env["UACLIENT_BEHAVE_BUILD_PR"] = "1"
+
+                # Inject tokens from credentials
+                for t in token:
+                    envvar = TOKEN_TO_ENVVAR[t]
+                    env[envvar] = credentials["token"].get(envvar)
+
+                # Inject cloud-specific variables from credentials
+                for cloud in ("azure", "aws", "gcp"):
+                    if cloud in p:
+                        for envvar in credentials[cloud]:
+                            env[envvar] = credentials[cloud].get(envvar)
+                # Tox command itself
+                command = "tox -e behave-{}-{}".format(
+                    p, series_version
+                ).split()
+
+                # Wip
+                if wip:
+                    command.extend(["--tags='wip'", "--stop"])
+
+                commands.append((command, env))
+
+    return commands
+
+
+@click.command()
+@click.option(
+    "-p",
+    "--platform",
+    type=click.Choice(PLATFORMS),
+    default=PLATFORMS,
+    multiple=True,
+    help="Platform to run the tests on (requires credentials file for the clouds)",
+)
+@click.option(
+    "-s",
+    "--series",
+    type=click.Choice(SERIES_TO_VERSION.keys()),
+    default=SERIES_TO_VERSION.keys(),
+    multiple=True,
+    help="Series to run the tests on",
+)
+@click.option(
+    "--token",
+    "-t",
+    type=click.Choice(TOKEN_TO_ENVVAR.keys()),
+    default=TOKEN_TO_ENVVAR.keys(),
+    multiple=True,
+    help="Tokens to use for the tests (require tokens in the credentials file)",
+)
+@click.option(
+    "--proposed",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run tests based on existing packages in the Proposed pocket",
+)
+@click.option(
+    "--check-version",
+    type=str,
+    help="Check for a specific version in the tests",
+)
+@click.option(
+    "--credentials-path",
+    "-c",
+    type=click.Path(dir_okay=False),
+    default="tools/ua-test-credentials.yaml",
+    help="Path to the file containing the credentials to run the test",
+)
+@click.option(
+    "-w",
+    "--wip",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run only tests with the @wip decorator",
+)
+def run_tests(
+    platform, series, proposed, check_version, token, credentials_path, wip
+):
+    commands = build_commands(
+        platform, series, proposed, check_version, token, credentials_path, wip
+    )
+    output_dir = "test-results"
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+    os.makedirs(output_dir)
```

Comment:
Drive-by thought: I'd rather not remove all previous test results. Can we instead create a new directory under `test-results` based on the datetime the tests are run maybe? Then we don't have to remove old test results and the new test results will get their own directory with a name that indicates the time they were run.